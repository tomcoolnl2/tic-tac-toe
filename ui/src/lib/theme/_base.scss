
@import url(//netdna.bootstrapcdn.com/font-awesome/3.2.1/css/font-awesome.css);

* {
    box-sizing: border-box;
    border: 0;
    margin: 0;
    padding: 0;
}

body {
    // background: var(--emerald-forest);
    background: blue;
    font-family: Arial, Helvetica, sans-serif;
    text-transform: uppercase;
    letter-spacing: 2px;
    color: var(--misty-sky);
    width: 100vw;
    height: 100vh; 
    overflow: hidden;
    display: grid;
    place-items: center;
}

sub {
    color: var(--misty-sky);
}

.screen {
    position: relative;
    perspective: 1000px;
}

.screen-inner {
    background: var(--midnight-blue);
    border-radius: 8px;
    position: relative;
    width: 340px;
    height: 472px;
    transition: transform 0.8s;
    transform-style: preserve-3d;
    margin: 20px;
}

.screen-inner.landscape {
    width: 580px;
    height: 342px;
        
    .board {
        width: 300px;
    }
}

.screen.game .screen-inner {
    transform: rotateY(180deg);
}

.screen-front,
.screen-back,
.screen-error {
    position: absolute;
    inset: 0;
    padding: 20px;
    backface-visibility: hidden;
}

.screen-front {
    margin: auto;
    width: 340px;
}

.screen-back {
    transform: rotateY(180deg);
}