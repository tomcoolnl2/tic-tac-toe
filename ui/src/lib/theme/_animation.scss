$speeds: (
	fast: 0.2s,
	normal: 0.3s,
	slow: 0.5s,
);
