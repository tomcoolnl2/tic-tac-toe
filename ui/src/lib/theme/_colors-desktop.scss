.desktop {
    --midnight-blue: #212F3C;
    --emerald-forest: #34495E;
    --aqua-glow: deeppink;
    --teal-surf: #AA004B;
}

