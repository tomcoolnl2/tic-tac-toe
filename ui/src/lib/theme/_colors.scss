:root {
	--sunshine-gold: #f2b237;
	--bronze-amber: #c78c1d;

	--background-color-primary: #192a32;
	--background-color-secondary: #203640;
	--background-color-tertiary: #0f212a;

	--theme-primary: #32c3c0;
	--theme-secondary: #238586;

	--light-primary: #a8bec9;
	--light-secondary: #5e7680;

	--error-primary: tomato;
	--error-secondary: #ffb397;
}
