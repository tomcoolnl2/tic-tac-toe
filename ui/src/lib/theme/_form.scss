.login-form {
	width: 280px;
}

.error-message {
	opacity: 0.8;
	border-radius: 4px;
	text-align: center;
	border: 2px solid var(--error-primary);
	padding: 10px;
	background: var(--error-secondary);
}
