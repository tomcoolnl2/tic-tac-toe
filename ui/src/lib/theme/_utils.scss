
@function generate-box-shadow($size: 3px, $color) {
    @return 0 #{$size} 0 -1px #{$color};
}