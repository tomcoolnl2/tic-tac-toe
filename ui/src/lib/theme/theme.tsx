import React from 'react';
import './styles.scss';

export enum Themes {
	WEB = 'web',
	DESKTOP = 'desktop',
}

export interface Props {
	theme?: Themes;
	children: React.ReactNode;
}

export const Theme: React.FC<Props> = ({ theme = Themes.WEB, children }) => {
	React.useEffect(() => {
		Object.values(Themes).map((theme) => document.body.classList.remove(theme));
		document.body.classList.add(theme);
	}, [theme]);

	return <main>{children}</main>;
};
