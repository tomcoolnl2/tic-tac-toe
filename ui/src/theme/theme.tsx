import './styles.scss';

export const Theme: React.FC<{ children: React.ReactNode }> = ({
	children,
}) => <main>{children}</main>;
