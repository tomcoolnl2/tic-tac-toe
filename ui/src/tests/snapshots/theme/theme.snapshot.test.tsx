import { render } from '@testing-library/react';
import { Theme } from '../../../lib/theme/theme';

describe('Theme screen component snapshot test', () => {
	it('should match the snapshot', () => {
		const { asFragment } = render(
			<Theme>
				<div />
			</Theme>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
