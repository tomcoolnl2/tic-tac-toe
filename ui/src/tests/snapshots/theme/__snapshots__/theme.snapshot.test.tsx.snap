// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme screen component snapshot test should match the snapshot 1`] = `
<DocumentFragment>
  <main>
    <div />
  </main>
</DocumentFragment>
`;
